### PR TITLE
feat: adds error state to read hooks

### PR DESCRIPTION
**Pull request by cesarenaldi**

### CODE REVIEW

File: examples/web-wagmi/src/profiles/components/ProfileSelector.tsx [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 import { invariant } from '../../utils';
 
-type SelectProfileProps = {
+type ProfileSelectorProps = {
   onProfileSelected: (profile: ProfileFragment | null) => void;
 };
 
-export function SelectProfile({ onProfileSelected }: SelectProfileProps) {
-  const { data, loading } = useExploreProfiles({ limit: 30 });
+export function ProfileSelector({ onProfileSelected }: ProfileSelectorProps) {
```
Comment (reecejohnson) [inline examples/web-wagmi/src/profiles/components/ProfileSelector.tsx:10]: This is a much better name, thanks!

File: packages/react/src/helpers.ts [hunk 2]
```diff
@@ -1,71 +1,101 @@
...
+      data,
+      error: undefined,
+      loading: false,
+    };
   }
 
-  if (data && !loading) {
+  if (error) {
     return {
+      data: undefined,
```
Comment (reecejohnson) [inline packages/react/src/helpers.ts:52]: 👍

**Review by reecejohnson — APPROVED** [review]
Very nice 💚

File: examples/web-wagmi/src/profiles/components/ProfileSelector.tsx (hunk 1 above)
Comment (cesarenaldi) [reply examples/web-wagmi/src/profiles/components/ProfileSelector.tsx:10]: oh, I didn't come up with it, I just used the file name.

**Pull request merged**